src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2025,29 +2025,13 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // - Pre-process text pasted (presumably from the clipboard)
     //   before sending it over the terminal's connection, converting
     //   Windows-space \r\n line-endings to \r line-endings
+    // - Also converts \n line-endings to \r line-endings
     void TermControl::_SendPastedTextToConnection(const std::wstring& wstr)
     {
-        // Some notes on this implementation:
-        //
-        // - std::regex can do this in a single line, but is somewhat
-        //   overkill for a simple search/replace operation (and its
-        //   performance guarantees aren't exactly stellar)
-        // - The STL doesn't have a simple string search/replace method.
-        //   This fact is lamentable.
-        // - This line-ending conversion is intentionally fairly
-        //   conservative, to avoid stripping out lone \n characters
-        //   where they could conceivably be intentional.
-
-        std::wstring stripped{ wstr };
-
-        std::wstring::size_type pos = 0;
-
-        while ((pos = stripped.find(L"\r\n", pos)) != std::wstring::npos)
-        {
-            stripped.replace(pos, 2, L"\r");
-        }
+        std::wregex multiLines{ LR"(((\r)?)(\n))" };
+        auto res = std::regex_replace(wstr, multiLines, L"\r");

```

Comment:
Same as above

Reply:
Thanks for all the suggestions! No longer using regex now :)